core/src/smartcontracts/isi/triggers/set.rs

```diff
@@ -39,7 +40,88 @@ pub enum Error {
 pub type Result<T, E = Error> = core::result::Result<T, E>;
 
 /// Type of action with pre-loaded executable.
-pub type LoadedAction<F> = Action<F, LoadedExecutable>;
+#[derive(Clone, Debug)]
+pub struct LoadedAction<F> {
+    executable: LoadedExecutable,
+    pub repeats: Repeats,
+    pub authority: AccountId,
+    pub filter: F,
+    pub metadata: Metadata,
+}
+
+pub trait LoadedActionTrait {
```

Comment:
I don't think there is any need to have this trait

Reply:
It's used because `triggers::Set` stores `Action`s in [specialized maps](https://github.com/hyperledger/iroha/blob/107545cf45af1d457649f1dac63b4f6f2fac02a7/core/src/smartcontracts/isi/triggers/set.rs#L134) but allows to inspect [them regardless of filter type](https://github.com/hyperledger/iroha/blob/107545cf45af1d457649f1dac63b4f6f2fac02a7/core/src/smartcontracts/isi/triggers/set.rs#L554)
I don't really understand why `Action` needs to be parametrized by filter type at all and not have an enum instead, but unless we change that, I don't think there's an easy way to get rid of this trait.

Reply:
after you have undrafted the PR, if you can remove this trait and replace trait objects with concrete types fine. If not, also fine. It's not a big deal

Reply:
Opened https://github.com/hyperledger/iroha/issues/3986